src/types/index.ts

```diff
@@ -59,6 +63,12 @@ export type RenderHookResult<
   Omit<TRenderer, keyof Renderer<TProps>> &
   AsyncUtils
 
+export type ServerRenderHookResult<
+  TProps,
+  TValue,
+  TRenderer extends ServerRenderer<TProps> = ServerRenderer<TProps>
+> = RenderHookResult<TProps, TValue, TRenderer>
```

Comment:
I introduced a few server render specific types here to help out in the tests.  The technically aren't used by the production code, but they don't hurt it either.